app/services/action_runner.py

```diff
@@ -47,35 +46,31 @@ async def execute_action(integration_id: str, action_id: str, config_overrides:
             content=jsonable_encoder({"detail": message}),
         )
 
-    # If "config" variable present in the request, we bypass the integration/config retrieval from the portal
-    if config_data:
-        config = config_data
-    else:
-        # Look for the configuration of the action being executed
-        action_config = find_config_for_action(
-            configurations=integration.configurations,
-            action_id=action_id
+    # Look for the configuration of the action being executed
+    action_config = find_config_for_action(
+        configurations=integration.configurations,
+        action_id=action_id
+    )
+    if not action_config and not config_overrides:
+        message = f"Configuration for action '{action_id}' for integration {str(integration.id)} " \
+                  f"is missing. Please fix the integration setup in the portal or include a config to override."
+        logger.error(message)
+        await publish_event(
+            event=IntegrationActionFailed(
+                payload=ActionExecutionFailed(
+                    integration_id=integration_id,
+                    action_id=action_id,
+                    error=f"Configuration missing for action '{action_id}'",
+                    config_data={"configurations": [i.dict() for i in integration.configurations]},
+                )
+            ),
+            topic_name=settings.INTEGRATION_EVENTS_TOPIC,
+        )
+        return JSONResponse(
+            status_code=status.HTTP_404_NOT_FOUND,
+            content=jsonable_encoder({"detail": message}),
         )
-        if not action_config:
-            message = f"Configuration for action '{action_id}' for integration {str(integration.id)} " \
-                      f"is missing. Please fix the integration setup in the portal."
-            logger.error(message)
-            await publish_event(
-                event=IntegrationActionFailed(
-                    payload=ActionExecutionFailed(
-                        integration_id=integration_id,
-                        action_id=action_id,
-                        error=f"Configuration missing for action '{action_id}'",
-                        config_data={"configurations": [i.dict() for i in integration.configurations]},
-                    )
-                ),
-                topic_name=settings.INTEGRATION_EVENTS_TOPIC,
-            )
-            return JSONResponse(
-                status_code=status.HTTP_404_NOT_FOUND,
-                content=jsonable_encoder({"detail": message}),
-            )
-        config = action_config.data
+    config = action_config.data if action_config else {}
```

Comment:
This might need to change a bit. If you look down at line 88 you'll see it's recording something different based on this code. It's been recording `config_data=action_config` but now will record either the nested data or the config_overrides.

Reply:
Thanks for checking that  @chrisdoehring. IMO it's good as it is now in this PR, but I may be missing something else.

It looks like the intention with the changes on this line `if action_config else {}` is to support the case where there isn't a configuration saved in the portal but there are config overrides. I imagine that that may be the case, for example, when the user is creating a connection and wants to test credentials before moving forward, or if we trigger a sub-action maybe, or in other cases if a configuration gets deleted for some reason.

The original intention on line 88 was to save the final configuration (base + overrides). Then if the final config triggers a parsing error, we can see the error in the activity logs and we can see the config value in the details for troubleshooting.